### PR TITLE
[server] Handle terminals for tasks that have already been closed

### LINKED ISSUE #5770: No logs when visiting a running prebuild
Reproduce:
 - start prebuild
 - go to prebuild page and follow logs
 - at a later point (during uploading image) reload page and see that there are no logs.

### PR DESCRIPTION
**Pull request by geropl**
## Description
On certain repositories (especially gitpod-io/gitpod) prebuild logs would not show in most cases while the prebuild was still running. The problem was caused by broken handling of terminals that have already finished.

## Related Issue(s)

Fixes #5770

## How to test
 1. clone gitpod-io/gitpod into your account
 2. go to https://gpl-5770-logs.staging.gitpod-dev.com/new and create a new team and import that project
 3. trigger a prebuild and view it's logs: should always display/stream

## Release Notes

```release-note
[prebuilds] fix prebuild logs with multiple tasks
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#5770*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/workspace/headless-log-service.ts [hunk 1]
```diff
@@ -104,11 +104,15 @@ export class HeadlessLogService {
...
             const terminalId = task.getTerminal();
-            if (terminalId === "") {
+            if (task.getState() === TaskState.OPENING) {
                 // this might be the case when there is no terminal for this task, yet.
                 // if we find any such case, we deem the workspace not ready yet, and try to reconnect later,
                 // to be sure to get hold of all terminals created.
-                throw new Error(`instance's ${wsi.id} task ${task.getId} has no terminal yet`);
+                throw new Error(`instance's ${wsi.id} task ${task.getId()} has no terminal yet`);
+            }
+            if (task.getState() === TaskState.CLOSED) {
```
Comment (geropl) [inline components/server/src/workspace/headless-log-service.ts:113]: @akosyakov This means we cannot reliably get logs from _all_ tasks from a running workspace. Is there a supervisor API that allows access to terminal of already closed tasks? If not, it'd be nice to add this to solve this issue.

Note that this is not super-relevant atm as we only show the output of a single terminal in the first place.

Comment (akosyakov) [reply components/server/src/workspace/headless-log-service.ts:113]: There is no terminal for closed tasks. It exists. If all terminal exists then prebuild is finished.

But each prebuild task has a file associated which is used later to show history on startup. Probably the task API could have a new method to retrieve content of such file.

Comment (geropl) [reply components/server/src/workspace/headless-log-service.ts:113]: We discussed offline, result in this ticket: https://github.com/gitpod-io/gitpod/issues/5922

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 154cf1b4421b5f003d9fb67e41d37a37b10cb625</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5918#issuecomment-930091456" title="LGTM">akosyakov</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#5770*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [akosyakov]
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5918#issuecomment-930091456" title="LGTM">akosyakov</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#5770*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5918#issuecomment-930106605" title="LGTM">akosyakov</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** after the PR has been reviewed.

Associated issue: *#5770*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5918#issuecomment-930106605" title="LGTM">akosyakov</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5918#issuecomment-930107168" title="Approved">geropl</a>*

Associated issue: *#5770*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**